Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvSpacepointsWriter.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/EventData/Measurement.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/GeometryHierarchyMap.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "ActsExamples/Digitization/DigitizationConfig.hpp"
+#include "ActsExamples/Digitization/SmearingAlgorithm.hpp"
+#include "ActsExamples/EventData/Cluster.hpp"
+#include "ActsExamples/EventData/Index.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Framework/WriterT.hpp"
+
+#include "ActsExamples/EventData/SimSpacePoint.hpp"
+
+#include <string>
+
+namespace ActsExamples {
+
+/// @class CsvSpacePointWriter
+///
+/// This writes one file per event containing information about the
+/// spacepoints
+///
+///     event000000001-sp.csv
+///     event000000002-sp.csv
+///     ...
+///
+/// Intrinsically thread-safe as one file per event.
+class CsvSpacepointsWriter final : public WriterT<SimSpacePointContainer> {
```

Comment:
This is a bit of a nitpick, but I think it would be more consistent with the other readers/writers if this was using the singular here.

```suggestion
class CsvSpacepointWriter final : public WriterT<SimSpacePointContainer> {
```